src/middlewared/middlewared/alembic/versions/21.MM/2021-05-25_22-00_afp_to_smb.py

```diff
@@ -0,0 +1,162 @@
+"""Convert AFP shares to SMB shares
+
+Revision ID: 5f4cab067712
+Revises: bd637e18fb0b
+Create Date: 2021-05-25 19:00:34.776266+00:00
+
+"""
+import json
+import textwrap
+
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = '5f4cab067712'
+down_revision = 'bd637e18fb0b'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    with op.batch_alter_table('sharing_cifs_share', schema=None) as batch_op:
+        batch_op.add_column(sa.Column('cifs_timemachine_quota', sa.Integer(), nullable=True))
+
+    op.execute("UPDATE sharing_cifs_share SET cifs_timemachine_quota = 0")
+
+    with op.batch_alter_table('sharing_cifs_share', schema=None) as batch_op:
+        batch_op.alter_column('cifs_timemachine_quota', existing_type=sa.INTEGER(), nullable=False)
+
+    conn = op.get_bind()
+    has_cifs_home = bool(conn.execute("SELECT * FROM sharing_cifs_share WHERE cifs_home = 1"))
+    disable_acl_if_trivial = []
+    for share in conn.execute("SELECT * FROM sharing_afp_share").fetchall():
+        cifs_auxsmbconf = [
+            "fruit:locking = netatalk",
+            "fruit:metadata = netatalk",
+            "fruit:resource = file",
+            "streams_xattr:prefix = user.",
+            "streams_xattr:store_stream_type = no",
+            "oplocks = no",
+            "level2 oplocks = no",
+            "strict locking = auto",
+        ]
```

Comment:
Rather than adding auxiliary parameters, perhaps we should add a column named something like `cifs_legacy_afp_compabilty` and then alter smb.conf generation based on that.

Reply:
These are pulled from `MULTI_PROTOCOL_AFP` purpose so migrated shares are identical to `MULTI_PROTOCOL_AFP` samba being created manually, is this a wrong approach?

Reply:
We don't want the auxiliary parameters to be written to the database. The preset only changes the running configuration. This allows us flexibility to fix without doing a migration and continues to move us away from auxiliary parameters.

I think that creating a new column in the cifs table with a name like `legacy_afp_compatibility` is better. Since we no longer have AFP, we can also migrate users away from having that preset as well (i.e. preset -> legacy_afp_compatibillity). Goal is to not have this option presented prominently and not write things to auxiliary parameters.

Reply:
So if I get you right you want to remove `MULTI_PROTOCOL_AFP` purpose and add a checkbox that does the same thing instead? Won't this introduce unnecessary conflicts (e.g. checkbox enabled and some non-trivial purpose selected)?